launch/launch/launcher.py

```diff
@@ -51,13 +57,35 @@ def add_launch_descriptor(self, launch_descriptor):
 
             self.task_descriptors.append(task_descriptor)
 
+    def interrupt_launch(self):
+        with self.loop_lock:
+            if self.loop is not None:
+                self.loop.call_soon_threadsafe(self.interrupt_launch_non_threadsafe)
+
+    def interrupt_launch_non_threadsafe(self):
+        self.interrupt_future.set_result(True)
+
+    def wait_on_launch_to_finish(self, timeout):
+        return self.launch_complete.wait(timeout)
+
+    def launch_is_running(self):
+        return not self.launch_complete.is_set()
+
+    def wait_on_processes_to_launch(self, timeout):
+        return self.processes_launched.wait(timeout)
+
+    def processes_are_launched(self):
```

Comment:
Is that a written rule? It was easier to read as `if [not] launcher.processes_are_launched:` rather than `if [not] launcher.are_processes_launched`, but I guess I'll change it: 1325b3d